saleor/discount/utils.py

```diff
@@ -74,14 +76,15 @@ def get_product_discounts(
     product: "Product",
     collections: Iterable["Collection"],
     discounts: Iterable[DiscountInfo],
-    channel: "Channel"
+    channel: "Channel",
+    variant_id: Optional[int] = None
```

Comment:
Why `variant_id` is optional here and `product` is required?

Reply:
connected with: https://github.com/mirumee/saleor/pull/8064#discussion_r714602215